### ISSUE TITLE
Setting additional seeds on existing datacenter causes a rolling restart

### ISSUE THREAD

**Issue by jsanda** [k8ssandra/k8ssandra-operator#67]
We have an e2e test that creates a 2-DC cluster. I observed with that test that the after dc2 became ready dc1 was undergoing a StatefulSet update which is effectively a rolling restart of Cassandra. This has to do with how/when the `AdditionalSeeds` property is configured.

`AdditionalSeeds` is set for dc2 when it is created because we already have the pod IPs for dc1. `AdditionalSeeds` is set for dc1 after dc2 becomes ready because it is only then that we have its pod IPs. In cass-operator the `GetConfigAsJSON` function creates a json representation of the Cassandra and JVM configuration which is then assigned in an env var for the config builder init container. That function has this:

```go
	seeds := []string{dc.GetSeedServiceName()}
	if len(dc.Spec.AdditionalSeeds) > 0 {
		seeds = append(seeds, dc.GetAdditionalSeedsServiceName())
	}
```

When the K8ssandraCluster controller sets `AdditionalSeeds` for dc1, the above if block is executed resulting in a change to the PodTemplateSpec of the StatefulSet, hence the StatefulSet update.

Aside from causing test failures with timeouts due to the rolling restart, it would be good to avoid this rolling restart if possible. We could add something to the CassandraDatacenter to let cass-operator know it should go ahead and create the additional seeds service even if there are no endpoints. This would allow us to avoid the StatefulSet update.  I am not a fan of doing this with a new property in the CRD because I feel like this is more about changing cass-operator behavior vs described the desired state of a Cassandra datacenter. I think an annotation on the CassandraDatacenter object would be better.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-782) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-782
┆Priority: Medium

**Comment by burmanm** [comment]
@jsanda I was thinking about another approach as well. What about creating the additional-seeds service, but not controlling it at all with cassdc.Spec.AdditionalSeeds?

What if we added a property to cass-operator that makes it not-controlling the additional-seeds service, but only use it when constructing? Thus, we could let the k8ssandra-operator manage the additional-seeds-service while cass-operator would simply ignore the additionalSeeds spec? 

Reduces the amount of reconciles in cass-operator and makes that seeds-service externally managed (even for some use cases outside k8ssandra-operator). 

There's actually no need for any new parameter even, I'll just change the Cassandra node creation to use the service's set seeds instead of Spec's.

**Comment by jsanda** [comment]
@burmanm this is much more in line with what I had in mind when we having the initial design discussions for k8ssandra-operator. 

Do you think it should be done in the k8ssandra cluster controller or elsewhere?

**Comment by burmanm** [comment]
Well, while there's no real dependency - it probably should happen in the K8ssandraCluster controller, since we don't otherwise know which remote clusters are part of the Cassandra cluster. Unless we use cass-operator's additionalServiceLabels  / annotations to add them.

**Comment by burmanm** [comment]
This was fixed in 1.8.0-rc.1
